src/collectors/processcheck/processcheck.py

```diff
@@ -0,0 +1,183 @@
+# coding=utf-8
+
+"""
+A Diamond collector that checks process defined in it's
+config file by matching them with their executable filepath or the process name.
+
+Example config file ProcessCheckCollector.conf
+
+```
+enabled=True
+ttl=150
+[process]
+[[postgres]]
+exe=^\/usr\/lib\/postgresql\/+d.+d\/bin\/postgres$
+name=^postgres,^pg
+
+```
+exe and name are both lists of comma-separated regexps.
```

Comment:
Why not a single regex which supports ORs? Why have multiple regexes?

Reply:
the configuration part is the same as ProcessResourcesCollector, this is so that there are some consistencies.

Reply:
Does it make sense for this, though? I don't like consistency just for the sake of consistency.

Reply:
for each check collector, there is an equivalent metric collector, so it's up to user on how they want to monitor it. so I'd keep the code and variable convention as close as possible.

Reply:
It's a separate config, correct? So even if the config format was consistent it'd just be for convenience. No reason for the code to be similar, though. Users don't interact with the code.

Reply:
ok, I'll update the variable names and see if I can simply the code a bit comparing to ProcessResourcesCollector.